cpp/src/io/orc/stripe_enc.cu

```diff
@@ -878,6 +892,7 @@ __global__ void __launch_bounds__(block_size)
             n = IntegerRLE<CI_DATA2, uint64_t, false, 0x3ff, block_size>(
               s, s->lengths.u64, s->nnz - s->numlengths, s->numlengths, flush, t, temp_storage.u64);
             break;
+          case DECIMAL:
```

Comment:
Decimal secondary is `signed integer`, may it is better to separate the Decimal from String 
```suggestion
          case DECIMAL:
            n = IntegerRLE<CI_DATA2, int32_t, true, 0x3ff, block_size>(
                s, s->lengths.i32, s->nnz - s->numlengths, s->numlengths, flush, t, temp_storage.i32);
            break;
             
```

Reply:
The signed values are already zigzag-d in case of decimal, so we do have unsigned values here. I can instead not zigzag, and separate this code out (and add i32 to `lenghts`). I would prefer the current code, given that the secondary stream values are converted to unsigned in all cases.

Reply:
Missed that one